crates/oxc_linter/src/rules/typescript/no_var_requires.rs

```diff
@@ -44,7 +42,10 @@ impl Rule for NoVarRequires {
         let AstKind::CallExpression(expr) = node.kind() else { return };
 
         if expr.is_require_call()
-            && !ctx.scopes().get_bindings(node.scope_id()).contains_key("require")
+            && ctx
+                .scopes()
+                .ancestors(node.scope_id())
+                .all(|scope_id| !ctx.scopes().get_bindings(scope_id).contains_key("require"))
```

Comment:
I think we have a method that indicates this identifier reference is undeclared? This looking up the scope chain seems expensive.